engine-src/Game/LambdaHack/Client/UI/HandleHumanGlobalM.hs

```diff
@@ -308,7 +308,7 @@ moveRunHuman initialStep finalGoal run runAhead dir = do
       cli {srunning = Just runParams}
     when runAhead $
       modifySession $ \cli ->
-        cli {slastPlay = (InGameMacro . map K.mkKM $ macroRun25) <> slastPlay cli}
+        cli {slastPlay = (KeyMacro . map K.mkKM $ macroRun25) <> slastPlay cli}
```

Comment:
I'm afraid here and in some other places we violate the newtype invariant, because the keys are not in reversed order here. I see I've set a trap for you, because the invariant can't be that the keys are reversed, too much code would need to be modified in a silly way. Only the invariant that keys are in correct order makes sense.

Reply:
Yup, that was silly, I didn't connected the threads: in `smacroBuffer` we keep macros in reversed order, but `slistPlay` is also of `Macro` type and yet before we push there any macro, we reverse it. I'll document this in both places.

Reply:
But hey, I think this invariant could be hold up if we reverse the macro right after we stop recording, not when repeating. Then it would be in the reverse order only with `Left` values.

Reply:
Indeed, the data invariant can be recovered if we ignore my earlier advice to move the reverse to the use site. I'm all for it. :)